lib/wicked/wlan.pm

```diff
@@ -0,0 +1,163 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base class for all WLAN related tests
+# Maintainer: cfamullaconrad@suse.com
+
+
+package wicked::wlan;
+
+use Mojo::Base 'wickedbase';
+use utils qw(random_string);
+use version_utils qw(is_sle);
+use registration qw(add_suseconnect_product);
+use utils qw(zypper_call);
+use testapi;
+
+has dhcp_enabled => 0;
+has eap_user     => 'tester';
+has eap_password => 'test1234';
+
+has netns_name   => 'wifi_ref';
+has ref_ifc      => 'wlan0';
+has ref_phy      => 'phy0';
+
+sub ref_ip {
+    my $self = shift;
+    my $type = $self->dhcp_enabled ? 'wlan_dhcp' : 'wlan';
+    return $self->get_ip(type => $type, is_wicked_ref => 1);
+}
+
+has sut_ifc => 'wlan1';
+has sut_phy => 'phy1';
+
+sub sut_ip {
+    my $self = shift;
+    my $type = $self->dhcp_enabled ? 'wlan_dhcp' : 'wlan';
+    return $self->get_ip(type => $type, is_wicked_ref => 0);
+}
+
+sub sut_hw_addr {
+    my $self = shift;
+    $self->{sut_hw_addr} //= $self->get_hw_address($self->sut_ifc);
+    return $self->{sut_hw_addr};
+}
+
+sub netns_exec {
+    my ($self, $cmd, @args) = @_;
+    $cmd = 'ip netns exec ' . $self->netns_name . ' ' . $cmd;
+    assert_script_run($cmd, @args);
+}
+
+sub netns_output {
+    my ($self, $cmd, @args) = @_;
+    $cmd = 'ip netns exec ' . $self->netns_name . ' ' . $cmd;
+    return script_output($cmd, @args);
+}
+
+sub ref_enable_DHCP_server {
+    my $self = shift;
+    $self->ref_disable_DHCP_server();
+    $self->dhcp_enabled(1);
+    $self->netns_exec(sprintf('dnsmasq --no-resolv --interface=%s --dhcp-range=%s,static --dhcp-host=%s,%s',
+            $self->ref_ifc, $self->sut_ip, $self->sut_hw_addr, $self->sut_ip));
+}
+
+sub ref_disable_DHCP_server {
+    my $self = shift;
+    $self->dhcp_enabled(0);
+    assert_script_run('test -e /var/run/dnsmasq.pid && kill $(cat /var/run/dnsmasq.pid) || true');
```

Comment:
any chance to do it more gracefully ? why we killing it ? also kill not give any guarantee that process will actually stop . it is sending TERM signal

Reply:
you mean by doing something like
`kill $pid; sleep 10; kill -9 $pid` ?

Reply:
this one too , but I was thinking more about **verification**  ( just brief googling point me to `kill -0 $pid` but any other way will work )

Reply:
I don't think it makes sense to check if the process is running and then killing it, if the kill command just return with error if the process doesn't.

I mean, we already have a `|| true` here. So I actually don't care if the process already died or not. 
For me, relaying on that pid file is enough.

Reply:
I will keep it.
Or I'm going to delete this method at all, cause technical there is no need.